pkg/skbn/abs.go

```diff
@@ -0,0 +1,236 @@
+package skbn
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+var err error
+
+func validateAbsPath(pathSplit []string) error {
+	if len(pathSplit) >= 1 {
+		return nil
+	}
+	return fmt.Errorf("illegal path: %s", filepath.Join(pathSplit...))
+}
+
+func initAbsVariables(split []string) (string, string, string) {
+	account := split[0]
+	container := split[1]
+	path := filepath.Join(split[2:]...)
+
+	return account, container, path
+}
+
+func getNewPipeline() (pipeline.Pipeline, error) {
+	accountName, accountKey := os.Getenv("AZURE_STORAGE_ACCOUNT"), os.Getenv("AZURE_STORAGE_ACCESS_KEY")
+
+	if len(accountName) == 0 || len(accountKey) == 0 {
+		err := errors.New("Either the AZURE_STORAGE_ACCOUNT or AZURE_STORAGE_ACCESS_KEY environment variable is not set")
+		return nil, err
+	}
+
+	credential, err := azblob.NewSharedKeyCredential(accountName, accountKey)
+
+	if err != nil {
+		return nil, err
+	}
+
+	po := azblob.PipelineOptions{
+		Retry: azblob.RetryOptions{
+			Policy:        azblob.RetryPolicyExponential,
+			MaxTries:      3,
+			TryTimeout:    time.Second * 3,
+			RetryDelay:    time.Second * 1,
+			MaxRetryDelay: time.Second * 3,
+		},
+	}
+
+	pl := azblob.NewPipeline(credential, po)
+
+	return pl, err
+}
+
+func getServiceURL(pl pipeline.Pipeline, accountName string) (azblob.ServiceURL, error) {
+	URL, err := url.Parse(
+		fmt.Sprintf("https://%s.blob.core.windows.net/", accountName))
+
+	surl := azblob.NewServiceURL(*URL, pl)
+	return surl, err
+}
+
+func getContainerURL(pl pipeline.Pipeline, accountName string, containerName string) (azblob.ContainerURL, error) {
+	URL, err := url.Parse(
+		fmt.Sprintf("https://%s.blob.core.windows.net/%s", accountName, containerName))
+
+	curl := azblob.NewContainerURL(*URL, pl)
+	return curl, err
+}
+
+func getBlobURL(curl azblob.ContainerURL, blob string) (azblob.BlockBlobURL, error) {
+	return curl.NewBlockBlobURL(blob), err
```

Comment:
function should only return the BlockBlobURL, no need to return error if nothing can be erroring inside...